src/Disciplina/Educator/Txs.hs

```diff
@@ -4,70 +4,27 @@
 module Disciplina.Educator.Txs
        ( PrivateTx (..)
        , PrivateTxId
-       , PrivateTxPayload (..)
-       , StudentTxMsg (..)
-       , EducatorTxMsg (..)
        , PrivateTxWitness (..)
        , PrivateTxAux (..)
        ) where
 
 import Universum
 
-import Disciplina.Core.Types (AssignmentId, CourseId, EducatorId, Grade, StudentId, Submission)
+import Disciplina.Core.Types (Grade, SignedSubmission (..))
 import Disciplina.Crypto (Hash, PublicKey, Signature)
 
 -- | Private transaction.
 data PrivateTx = PrivateTx
-    { _ptxStudentId  :: !StudentId
-    -- ^ Every transaction relates to one particular student.
-    , _ptxCourseId   :: !CourseId
-    -- ^ Every transaction relates to one particular course.
-    , _ptxEducatorId :: !EducatorId
-    -- ^ Included to simplify educator signature checking
-    -- on student's side.
-    , _ptxPayload    :: !PrivateTxPayload
-    -- ^ Actual contents of transaction.
+    { _ptxSignedSubmission  :: !SignedSubmission
+    -- ^ Every transaction contains one signed student submission
+    , _ptxGrade             :: !Grade
+    -- ^ Grade for this submission
+    , _ptxTime              :: !Integer
```

Comment:
Why not `UTCTime`, for example? If it's stored in DB as an integer timestamp, it doesn't mean we cannot have more semantically valid type in Haskell.